### PR TITLE
Allow HTTP Headers not in spec

### PR DESCRIPTION
**Pull request by hardillb**
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Proposed changes

Allow the relaxing of HTTP Header parsing

potential fix for #3772

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

Need to think about how to add tests for "broken" headers when using express as the test http server

### CODE REVIEW

**Review by Steve-Mcl — APPROVED** [review]
Discussed with Ben - looks ok for me.

File: packages/node_modules/@node-red/nodes/locales/en-US/messages.json [hunk 1]
```diff
@@ -554,7 +554,8 @@
         },
         "status": {
             "requesting": "requesting"
-        }
+        },
+        "insecureHTTPParser": "Lenient HTTP Header Parsing"
```
Comment (knolleary) [inline packages/node_modules/@node-red/nodes/locales/en-US/messages.json:558]:
```suggestion
        "insecureHTTPParser": "Disable strict HTTP parsing"
```

**Comment by Steve-Mcl** [comment]
@hardillb does this need backport to v2.x?

**Comment by knolleary** [comment]
This is borderline between new feature and bug fix. Given we haven't had lots of complaints about it, I'm not inclined to make more work for ourselves.

**Comment by hardillb** [comment]
But the change that triggers this is in nodejs 14/16/18 so is likely to hit new deployments of 2.2.2 on latest nodejs

**Comment by knolleary** [comment]
True - okay @Steve-Mcl - backport it.

**Comment by hardillb** [comment]
Which will include 2.2.3 docker release as it will pick up the new node point releases

**Comment by Steve-Mcl** [comment]
need to merge this & fetch to cherry pick - ok to merge first?

**Comment by knolleary** [comment]
Was just waiting for CI to finish. Merged

**Pull request merged**